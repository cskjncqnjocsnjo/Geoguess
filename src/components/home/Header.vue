<template>
    <div>
        <v-app-bar class="header" height="100">
            <img class="header__logo" src="@/assets/geoguessLogo.png" />

            <div class="flex-grow-1"></div>

            <nav class="header__nav">
                <v-btn text @click="historyDialog = true" id="historyBtn">{{
                    $t('Home.historyBtn')
                }}</v-btn>
                <div class="header__nav__btns">
                    <v-btn text @click="aboutDialog = true" id="aboutBtn">
                        <v-icon size="30">mdi-help-circle</v-icon>
                    </v-btn>
                    <v-menu>
                        <template v-slot:activator="{ on }">
                            <v-btn text v-on="on" id="languageBtn">
                                <v-icon size="30">mdi-translate</v-icon>
                            </v-btn>
                        </template>
                        <v-list id="menuLanguage">
                            <v-list-item
                                v-for="(language, index) in languages"
                                :key="index"
                                @click="switchLanguage(language.value)"
                            >
                                <v-list-item-title>{{
                                    language.text
                                }}</v-list-item-title>
                            </v-list-item>
                        </v-list>
                    </v-menu>
                </div>
            </nav>
            <v-dialog v-model="historyDialog">
                <History :history="history" />
            </v-dialog>
            <v-dialog v-model="aboutDialog">
                <About />
            </v-dialog>
        </v-app-bar>
    </div>
</template>
<script>
import History from '@/components/History';
import About from '@/components/home/About';
import { languages } from '../../lang';

export default {
    components: {
        History,
        About,
    },
    data() {
        return {
            drawer: false,
            historyDialog: false,
            aboutDialog: false,
            history: localStorage.getItem('history')
                ? JSON.parse(localStorage.getItem('history'))
                : [],
            languages,
        };
    },
    methods: {
        switchLanguage(language) {
            this.$i18n.locale = language;
            this.$vuetify.lang.current = language;
            this.saveLanguage(language);
        },
        saveLanguage(language) {
            localStorage.setItem('language', language);
        },
    },
};
</script>
<style lang="scss" scoped>
.header {
    padding: 0 5%;
    background-color: #f1e9d6 !important;
    nav.header__nav {
        display: flex;
        & > div {
            margin: 0 1.5rem;
        }
    }
    .v-btn {
        font-size: 1.2rem;
    }
    .header__logo {
        height: 6rem;
        width: auto;
    }
}

@media (max-width: 660px) {
    .header {
        .header__logo {
            display: none;
        }
    }
}
</style>
