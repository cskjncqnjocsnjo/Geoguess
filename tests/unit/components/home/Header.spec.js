import Header from '@/components/home/Header.vue';
import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import appInit from '../../utils/appInit';

const args = appInit(createLocalVue());

describe('Header.vue', () => {
    it('open dialog hist', () => {
        const wrapper = mount(Header, args);
        const buttonHistory = wrapper.find('#historyBtn');
        expect(buttonHistory.exists()).toBe(true);
        expect(wrapper.vm.historyDialog).toEqual(false);
        buttonHistory.trigger('click');
        expect(wrapper.vm.historyDialog).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });
    it('open dialog about', () => {
        const wrapper = mount(Header, args);

        const aboutBtn = wrapper.find('#aboutBtn');
        expect(wrapper.vm.aboutDialog).toEqual(false);
        aboutBtn.trigger('click');
        expect(wrapper.vm.aboutDialog).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('switchLanguage method', () => {
        const wrapper = shallowMount(Header, args);
        expect(localStorage.getItem('language')).toBeNull();
        wrapper.vm.switchLanguage('en');
        expect(localStorage.getItem('language')).toEqual('en');
    });

    it('test localSotrage history method', () => {
        localStorage.setItem(
            'history',
            '[{"multiplayer":false,"date":"2020-06-14T15:10:14.579Z","timeLimitation":0,"rounds":[{"guess":{"lat":49.35629642234583,"lng":-73.3876235},"position":{"lat":46.32492404792541,"lng":-74.2128121666204},"distance":343031},{"guess":{"lat":20.966226136901,"lng":105.91472350868774},"position":{"lat":16.45610423382063,"lng":107.5978548200058},"distance":532484},{"guess":{"lat":-1.9789104624962186,"lng":-64.70788738160854},"position":{"lat":-8.47747465683049,"lng":-70.14934638834765},"distance":941702},{"guess":{"lat":55.48785435635061,"lng":-1.5939715638887852},"position":{"lat":58.51518485516467,"lng":-6.260475420814613},"distance":439803},{"guess":{"lat":39.96452059424641,"lng":-100.91667319052173},"position":{"lat":42.04582308041186,"lng":-101.0496153025084},"distance":231958}],"score":2488978}]'
        );
        const wrapper = shallowMount(Header, args);
        expect(wrapper.vm.history).toHaveLength(1);
        expect(wrapper.vm.history[0].score).toEqual(2488978);
    });
});
