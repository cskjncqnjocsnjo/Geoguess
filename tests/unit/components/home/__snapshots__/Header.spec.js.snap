// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue open dialog about 1`] = `
<div>
  <header class="header v-sheet theme--light v-toolbar v-app-bar" style="height: 100px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 100px;"><img src="@/assets/geoguessLogo.png" class="header__logo">
      <div class="flex-grow-1"></div>
      <nav class="header__nav"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="historyBtn"><span class="v-btn__content">HISTORY</span></button>
        <div class="header__nav__btns"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="aboutBtn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-help-circle theme--light" style="font-size: 30px;"></i></span></button>
          <div class="v-menu"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="languageBtn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-translate theme--light" style="font-size: 30px;"></i></span></button>
            <!---->
          </div>
        </div>
      </nav>
      <div role="dialog" class="v-dialog__container">
        <!---->
      </div>
      <div role="dialog" class="v-dialog__container">
        <!---->
      </div>
    </div>
  </header>
</div>
`;

exports[`Header.vue open dialog hist 1`] = `
<div>
  <header class="header v-sheet theme--light v-toolbar v-app-bar" style="height: 100px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 100px;"><img src="@/assets/geoguessLogo.png" class="header__logo">
      <div class="flex-grow-1"></div>
      <nav class="header__nav"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="historyBtn"><span class="v-btn__content">HISTORY</span></button>
        <div class="header__nav__btns"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="aboutBtn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-help-circle theme--light" style="font-size: 30px;"></i></span></button>
          <div class="v-menu"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default" id="languageBtn"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-translate theme--light" style="font-size: 30px;"></i></span></button>
            <!---->
          </div>
        </div>
      </nav>
      <div role="dialog" class="v-dialog__container">
        <!---->
      </div>
      <div role="dialog" class="v-dialog__container">
        <!---->
      </div>
    </div>
  </header>
</div>
`;
